### PR TITLE
[TR] PIM-4513: Send a draft for approval

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-4513

| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N
| CI currently passed? | Y
| Tests pass?          | 
| Scenarios pass?      | 
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | -
| Fixed tickets        | PIM-4513
| Doc PR               |
| Related CE PR        | -

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js [hunk 1]
```diff
@@ -41,29 +41,40 @@ define(
...
                 this.listenTo(this.getFormModel(), 'change', this.render);
                 this.listenTo(mediator, 'entity:form:edit:update_state', this.render);
+                this.listenTo(mediator, 'product:action:post_update', this.collectAndRender);
+                this.listenTo(mediator, 'pim_enrich:form:state:confirm', this.onConfirmation);
 
-                mediator.on('product:action:post_update', _.bind(function (data) {
-                    this.state = JSON.stringify(data);
-                    this.render();
-                }, this));
+                $(window).on('beforeunload', _.bind(this.beforeUnload, this));
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js]: Are you sure about that ?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js [hunk 2]
```diff
@@ -72,17 +83,31 @@ define(
...
+            /**
+             * Callback triggered on beforeunload event
+             */
             beforeUnload: function () {
                 if (this.hasModelChanged()) {
                     return this.confirmationMessage;
                 }
             },
+
             linkClicked: function (event) {
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/state.js]: no doc ?

**Comment by sumbobyboys** [comment]
Nice PR ! GTM after discussion/fix :)

**Comment by jmleroux** [comment]
GTM

**Pull request merged**